src/validation/validation.cpp

```diff
@@ -2724,29 +2728,23 @@ void UpdateTip(CBlockIndex *pindexNew)
         CheckAndAlertUnknownVersionbits(chainParams, chainActive.Tip());
     }
 
-    if (IsNov152018Scheduled()) // Set the global variables based on the fork state of the NEXT block
+    // Set the global variables based on the fork state of the NEXT block
+    // Always allow overwite of enableCanonicalTxOrder but for regtest)
+    if (AreWeOnBCHChain() && IsNov152018Activated(chainParams.GetConsensus(), chainActive.Tip()) &&
+        chainParams.NetworkIDString() != "regtest")
     {
-        if (IsNov152018Enabled(chainParams.GetConsensus(), pindexNew))
-        {
-            enableCanonicalTxOrder = true;
-        }
-        else
-        {
-            enableCanonicalTxOrder = false;
-        }
+        enableCanonicalTxOrder = AreWeOnBCHChain();
     }
```

Comment:
again you took out ht enablecanonical=false

Reply:
ditto